src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -194,7 +193,7 @@ def apply(
         enabled_transformations: Optional[Transformation] = None,
         # allow sym references without a symbol declaration, mostly for testing
         allow_undeclared_symbols: bool = False,
-        uids: Optional[eve_utils.UIDGenerator] = None,
+        uids: utils.IDGeneratorPool,
```

Comment:
Another option to the fixtures would be:
```suggestion
        uids: utils.IDGeneratorPool | None,
```
and then in the apply
```python
uids = uids or utils.IDGeneratorPool()
```
This would ensure explicitness in the tests, but avoid the countless duplication of `utils.IDGeneratorPool` calls.